server/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java

```diff
@@ -176,4 +177,44 @@ void givenTwoTaskManagerATaskSubmittedOnOneCouldBeRunOnTheOther() throws Interru
         assertThat(executionDetails.getSubmittedNode()).isEqualTo(HOSTNAME_2);
         assertThat(executionDetails.getRanNode()).contains(HOSTNAME);
     }
+
+    private List<TaskId> extractIds(List<TaskExecutionDetails> tasks) {
+        return tasks.stream()
+            .map(TaskExecutionDetails::getTaskId)
+            .collect(Guavate.toImmutableList());
+    }
+
+    private boolean allTasksAreCompleted(List<TaskExecutionDetails> tasks) {
+        List<TaskExecutionDetails> completedTasks = tasks.stream()
+            .filter(task -> task.getStatus().equals(TaskManager.Status.COMPLETED))
+            .collect(Guavate.toImmutableList());
+
+        return completedTasks.size() == tasks.size();
+    }
+
+    @Test
+    void givenTwoTaskManagerAndATaskRanPerTaskManagerListingThemOnEachShouldShowBothTasks() {
+        TaskManager taskManager1 = taskManager(HOSTNAME);
+        TaskManager taskManager2 = taskManager(HOSTNAME_2);
+
+        TaskId taskId1 = taskManager1.submit(new CompletedTask());
+        TaskId taskId2 = taskManager2.submit(new CompletedTask());
+
+        int expectedNumberOfTasks = 2;
+
+        Awaitility.await()
+            .atMost(Duration.ONE_SECOND)
+            .pollInterval(100L, TimeUnit.MILLISECONDS)
+            .untilAsserted(() -> {
```

Comment:
it's usually a bad idea to put the actual assertion into an until. Usually, we wait for some events then we do the assertions.

Reply:
At first i used a condition into an until. But without assertion after that.
From a previous comment i refactored it with untilAsserted.
I don't see the point in rewriting it in this particular case. As the conditions needed  to do the assertions correspond exactly to the assertions.

Reply:
Can you tell me what's the output of the test when it fails?

Reply:
Yes for example if i make the test fail on the assertion comparing the size of the list.
I have the line and actual/expected value.

```
Expecting:
 <2>
to be equal to:
 <3>
but was not.
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at org.apache.james.task.eventsourcing.distributed.DistributedTaskManagerTest.lambda$givenTwoTaskManagerAndATaskRanPerTaskManagerListingThemOnEachShouldShowBothTasks$5(DistributedTaskManagerTest.java:215)
```